internal/providers/azure/azure.go

```diff
@@ -66,29 +64,51 @@ func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
 // FetchFromOvfDevice has the return signature of platform.NewFetcher. It is
 // wrapped by this and AzureStack packages.
 func FetchFromOvfDevice(f *resource.Fetcher, ovfFsTypes []string) (types.Config, report.Report, error) {
-	devicePath := filepath.Join(distro.DiskByIDDir(), configDeviceID)
-
+	var rawConfig []byte
+	var err error
 	logger := f.Logger
-	logger.Debug("waiting for config DVD...")
-	waitForCdrom(logger, devicePath)
 
-	fsType, err := checkOvfFsType(logger, devicePath, ovfFsTypes)
-	if err != nil {
-		return types.Config{}, report.Report{}, err
+	for _, ovfFsType := range ovfFsTypes {
+		device, err := execUtil.GetBlockDevices(ovfFsType)
+		if err != nil {
+			return types.Config{}, report.Report{}, fmt.Errorf("failed to retrieve block devices with FSTYPE=%q: %v", ovfFsType, err)
+		}
+		for len(device) > 0 {
+			for i, dev := range device {
+				// verify that this is a CD-ROM device
+				if isCdromPresent(logger, dev) {
+					rawConfig, err = getRawConfig(f, dev, ovfFsType)
+					if len(rawConfig) >= 0 {
```

Comment:
`len(rawConfig)` can't be negative, so this is always true.  You can do what you were trying to do here by following the convention of handling the error case first.